src/Workspaces/Core/Portable/NamingStyles/EditorConfig/EditorConfigNamingStyleParser.cs

```diff
@@ -85,5 +145,114 @@ where kvp.Key.Trim().StartsWith("dotnet_naming_rule.", StringComparison.Ordinal)
+                var secondIsSubset = FirstIsSubset(in y, in x);
+                if (firstIsSubset)
+                {
+                    return secondIsSubset ? 0 : -1;
+                }
+                else
+                {
+                    return secondIsSubset ? 1 : 0;
+                }
+            }
+
+            /// <summary>
+            /// Determines if <paramref name="x"/> matches a subset of the symbols matched by <paramref name="y"/>. The
+            /// implementation determines which properties of <see cref="NamingRule"/> are considered for this
+            /// evaluation. The subset relation does not necessarily indicate a proper subset.
+            /// </summary>
+            /// <param name="x">The first naming rule.</param>
+            /// <param name="y">The second naming rule.</param>
+            /// <returns><see langword="true"/> if <paramref name="x"/> matches a subset of the symbols matched by
+            /// <paramref name="y"/> on some implementation-defined properties; otherwise, <see langword="false"/>.</returns>
+            protected abstract bool FirstIsSubset(in NamingRule x, in NamingRule y);
+        }
+
+        private sealed class NamingRuleAccessibilityListComparer : NamingRuleSubsetComparer
+        {
+            internal static readonly NamingRuleAccessibilityListComparer Instance = new NamingRuleAccessibilityListComparer();
+
+            private NamingRuleAccessibilityListComparer()
+            {
+            }
+
+            protected override bool FirstIsSubset(in NamingRule x, in NamingRule y)
+            {
+                foreach (var accessibility in x.SymbolSpecification.ApplicableAccessibilityList)
+                {
+                    if (!y.SymbolSpecification.ApplicableAccessibilityList.Contains(accessibility))
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+        }
+
+        private sealed class NamingRuleModifierListComparer : NamingRuleSubsetComparer
+        {
+            internal static readonly NamingRuleModifierListComparer Instance = new NamingRuleModifierListComparer();
+
+            private NamingRuleModifierListComparer()
+            {
+            }
+
+            protected override bool FirstIsSubset(in NamingRule x, in NamingRule y)
+            {
+                // Since modifiers are "match all", a subset of symbols is matched by a superset of modifiers
+                foreach (var modifier in y.SymbolSpecification.RequiredModifierList)
+                {
+                    if (modifier.ModifierKindWrapper == SymbolSpecification.ModifierKindEnum.IsStatic
+                        || modifier.ModifierKindWrapper == SymbolSpecification.ModifierKindEnum.IsReadOnly)
+                    {
+                        if (x.SymbolSpecification.RequiredModifierList.Any(x => x.ModifierKindWrapper == SymbolSpecification.ModifierKindEnum.IsConst))
```

Comment:
Is it easier just writing RequiredModifierList.Contains(new ModifierKind(ModifierKindEnum.Const)) or something? I'll admit it tripped me up that here we're doing an Any and below we're doing a Contains, but that's largely because this whole wrapping thing is...icky. (And not your fault.)